artiq/coredevice/mirny.py

```diff
@@ -140,3 +140,144 @@ def write_ext(self, addr, length, data):
         if length < 32:
             data <<= 32 - length
         self.bus.write(data)
+
+
+class Almazny:
+    """
+    Almazny (High frequency mezzanine board for Mirny)
+
+    :param mirny - Mirny device Almazny is connected to
+    """
+
+    # pin map
+    SER_MOSI = 0
+    SER_CLK = 1
+    REG_LATCH_BASE = 2
+    REG_CLEAR = 6
+
+    mezzio_reg = 0x03  # register addr for Mezz_IO 0~7
+
+    def __init__(self, dmgr, host_mirny):
+        self.mirny = dmgr.get(host_mirny)
+        self.mezz_data = 0
+
+    @kernel
+    def init(self):
+        self._send_mezz_data([
+            (self.REG_CLEAR, 1),
+            (self.REG_LATCH_BASE+0, 0),
+            (self.REG_LATCH_BASE+1, 0),
+            (self.REG_LATCH_BASE+2, 0),
+            (self.REG_LATCH_BASE+3, 0),
+            (self.SER_CLK, 0)
+            ])
+
+    @kernel
+    def reg_set(self, reg_i, attin1=1, attin2=1, attin3=1, attin4=1, attin5=1, attin6=1, output_on=0):
+        """
+        Sets the data on chosen shift register.
+        :param reg_i - index of the register [1-4]
+        :param attin[1-6] - attenuator on {0, 1} 
+        :attin1 - 16dB attenuator, att6 - 0.5dB
```

Comment:
Use an integer that represents the raw ("machine units") 6-bit value loaded into the attenuator. Write conversion functions to/from machine units. See how it's done on Mirny and Urukul where the situation is exactly the same.